### PR TITLE
Remove and fold unused code

### LINKED ISSUE #1926: Delete unused functions from OMRLinkage.hpp
The following functions in `compiler/codegen/OMRLinkage.hpp` are not used in OMR nor any known downstream project.  They appear to be legacy code and can be deleted outright.

```
   virtual TR_GlobalRegisterNumber getFormalParameterGlobalRegister(TR::ParameterSymbol *sym) { return -1; }

   virtual TR::SymbolReference *getCachedPrivateStaticSymRef()  { return NULL; }
   virtual bool safeToUseCachedPrivateStaticSymRef()             { return false; }

   virtual bool usedCachedStaticSymRef()                         { return false; }

   virtual uint8_t getCodeNum()
      {
      TR_ASSERT(0, "getCodeNum has to be implemented for this linkage\n");
      return 0;
      }

   virtual uint8_t getDataNum()
      {
      TR_ASSERT(0, "getDataNum has to be implemented for this linkage\n");
      return 0;
      }

   virtual uint8_t getStatNum()
      {
      TR_ASSERT(0, "getStatNum has to be implemented for this linkage\n");
      return 0;
      }

  virtual bool preserveForInlining(uint8_t)
      {
      return true;
      }

   virtual TR::SymbolReference* getCachedStaticSymRef() { return NULL; }
   virtual TR::SymbolReference* getCached512KStaticSymRef(uint32_t index) { return NULL; }
   virtual TR::SymbolReference* getCachedCRENTSymRef() { return NULL; }

#define DEFAULT_NUM_WSA_BUCKETS 1
   virtual uint32_t getNumWSABuckets() { return DEFAULT_NUM_WSA_BUCKETS; }
```

### LINKED ISSUE #1929: Remove and fold getMainBodyLogicalParameterList from OMRLinkage
The function `getMainBodyLogicalParameterList` in `OMR::Linkage` is not used nor overridden in OMR or any known downstream project.  The function should be eliminated and all code that currently calls it should be folded accordingly assuming it returns `NULL`.

### LINKED ISSUE #1931: Remove and fold isAddressOfPrivateStaticSymRef from OMRLinkage
The function `isAddressOfPrivateStaticSymRef` in `OMR::Linkage` only returns `false` and is never overridden in either OMR or any known downstream project on any architecture.  Remove it and fold code assuming it always returns `false`.

### LINKED ISSUE #1932: Remove and fold isAlwaysInlined from OMRLinkage
The function `isAlwaysInlined` in `OMR::Linkage` only returns `false` and is never overridden in either OMR or any known downstream project on any architecture.  Remove it and fold code that consumes it assuming it always returns `false`.

### LINKED ISSUE #1933: Remove and fold isRecognizedBuiltin from OMRLinkage
The function `isRecognizedBuiltin` in `OMR::Linkage` only returns `false` and is never overridden in either OMR or any known downstream project on any architecture.  Remove it and fold code that consumes it assuming it always returns `false`.

### LINKED ISSUE #1934: Remove and fold getInRegisterNumberForParameter from OMRLinkage
The function `getInRegisterNumberForParameter` in `OMR::Linkage` only returns `-1` and is never overridden in either OMR or any known downstream project on any architecture.  Remove it and fold code that consumes it assuming it always returns `-1`.

### LINKED ISSUE #1935: Remove and fold getNumberOfAllocatedOutgoingArgumentAreas from OMRLinkage
The function `getNumberOfAllocatedOutgoingArgumentAreas` in `OMR::Linkage` only returns `0` and is never overridden in either OMR or any known downstream project on any architecture.  Remove it and fold code that consumes it assuming it always returns `0`.

### LINKED ISSUE #1936: Remove and fold isArgumentListSymbol from OMRLinkage
The function `isArgumentListSymbol` in `OMR::Linkage` only returns `false` and is never overridden in either OMR or any known downstream project on any architecture.  Remove it and fold code that consumes it assuming it always returns `false`.

### PR DESCRIPTION
**Pull request by shivammittal99**
Remove unused functions.

Closes #1926, closes #1929, closes #1931, closes #1932, closes #1933, closes #1934, closes #1935, closes #1936

### CODE REVIEW

**Comment by mgaudet** [comment]
Thanks so much for your contribution. We're really excited for these changes. 

I think we'd really like to see you break apart these changes into one pull request per issue. The reason is that it makes it much easier to review if the changes are split apart.

I know that this means you'll be turning one pull request into eight, but I think the end result is better. 

This will also give you the opportunity to write a more specialized commit message for each change. Feel free to read our CONTRIBUTING.md for some suggestions about writing good commit messages (they needn't be long, but should accurately describe the change in the title line).  

I hope this doesn't discourage you, and if you need any help or advice on splitting the changes apart, please feel free to ask more questions.

**Comment by charliegracie** [comment]
genie-omr build all

File: compiler/il/symbol/OMRResolvedMethodSymbol.cpp [hunk 1]
```diff
@@ -332,7 +332,7 @@ OMR::ResolvedMethodSymbol::getLogicalParameterList(TR::Compilation *comp)
    {
    if (comp->getMethodSymbol() == self())
       {
-      List<TR::ParameterSymbol>* l = comp->cg()->getLinkage()->getMainBodyLogicalParameterList();
+      List<TR::ParameterSymbol>* l = NULL;
```
Comment (mgaudet) [inline compiler/il/symbol/OMRResolvedMethodSymbol.cpp]: I think there's further folding that could be done here, given that literally the next check is `if (l == null)`

File: compiler/codegen/CodeGenRA.cpp [hunk 2]
```diff
@@ -1606,12 +1606,6 @@ OMR::CodeGenerator::pickRegister(TR_RegisterCandidate     *rc,
                      callNode = node->getFirstChild();
                      }
 
-                  if (callNode &&
```
Comment (mgaudet) [inline compiler/codegen/CodeGenRA.cpp]: Is it just me, or can you now edxtend the above block further down combining the block starting at 1607 with the block starting at 1609 (New file numbers)

File: compiler/codegen/CodeGenRA.cpp [hunk 3]
```diff
@@ -2130,7 +2130,7 @@ OMR::CodeGenerator::findCoalescenceRegisterForParameter(TR::Node *callNode, TR_R
       {
       int32_t paramSymRefNum = paramNode->getSymbolReference()->getReferenceNumber();
       *isUnpreferred = !(paramSymRefNum == rc->getSymbolReference()->getReferenceNumber());
-      TR_GlobalRegisterNumber candidateRegister = self()->getLinkage()->getInRegisterNumberForParameter(callNode, childIndex);
+      TR_GlobalRegisterNumber candidateRegister = -1;
```
Comment (mgaudet) [inline compiler/codegen/CodeGenRA.cpp]: This can get collapsed to `return -1`

File: compiler/codegen/CodeGenRA.cpp (hunk 2 above)
Comment (0xdaryl) [reply compiler/codegen/CodeGenRA.cpp]: Yes.  The only way `callNode` can be set to something non-NULL is on L1606.  Please add this folding opportunity to this commit.

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 4]
```diff
@@ -1474,7 +1474,7 @@ void
 OMR::Z::CodeGenerator::doInstructionSelection()
    {
 
-   _outgoingArgLevelDuringTreeEvaluation = self()->getLinkage()->getNumberOfAllocatedOutgoingArgumentAreas();
+   _outgoingArgLevelDuringTreeEvaluation = 0;
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: I believe this field and its accessors can be deleted as well as they are not used anywhere.  Please remove.

File: compiler/optimizer/OMRSimplifierHandlers.cpp [hunk 5]
```diff
@@ -5476,8 +5476,7 @@ TR::Node *indirectLoadSimplifier(TR::Node * node, TR::Block * block, TR::Simplif
       return resultNode;
 
    TR::Node *firstChild = node->getFirstChild();
-   if (firstChild->getOpCodeValue() == TR::loadaddr &&
-       !s->comp()->cg()->getLinkage()->isArgumentListSymbol(firstChild->getSymbolReference()->getSymbol(), s->comp())) // argList symbol can grow later on
+   if (firstChild->getOpCodeValue() == TR::loadaddr) // argList symbol can grow later on
```
Comment (0xdaryl) [inline compiler/optimizer/OMRSimplifierHandlers.cpp]: Please remove the comment too.

File: compiler/optimizer/OMRSimplifierHandlers.cpp [hunk 6]
```diff
@@ -5607,9 +5606,7 @@ TR::Node *indirectStoreSimplifier(TR::Node * node, TR::Block * block, TR::Simpli
       return NULL;
       }
 
-   if (
-       firstChild->getOpCodeValue() == TR::loadaddr &&
-       !s->comp()->cg()->getLinkage()->isArgumentListSymbol(firstChild->getSymbolReference()->getSymbol(), s->comp())) // argList symbol can grow later on
+   if (firstChild->getOpCodeValue() == TR::loadaddr) // argList symbol can grow later on
```
Comment (0xdaryl) [inline compiler/optimizer/OMRSimplifierHandlers.cpp]: Please remove the comment.

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 7]
```diff
@@ -547,8 +547,7 @@ OMR::Z::CodeGenerator::CodeGenerator()
      _ccInstruction(NULL),
      _previouslyAssignedTo(self()->comp()->allocator("LocalRA")),
      _bucketPlusIndexRegisters(self()->comp()->allocator()),
-     _currentDEPEND(NULL),
-     _outgoingArgLevelDuringTreeEvaluation(0)
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: Please shorten the commit title for this commit (< 70 chars per the CG).

**Comment by 0xdaryl** [comment]
genie-omr build all

**Pull request merged**